### PR TITLE
#164198137 setup testing

### PR DESCRIPTION
**Pull request by davdwhyte87**
**What does this PR do?**
This PR adds unit testing mechanism and end to end testing  to the application

**Tasks to be completed?**
- install jest
- install enzyme
- write test
- structure test file system
- install and  configure Cypress
- configure circle ci with Cypress tests

**How can this be manually tested?**
- clone the repo
- install `node js, yarn`
- run `yarn install` to install dependencies
- run `yarn test` to run tests

**Relevant PT stories?**
`64198137`

### CODE REVIEW

**Review by fejiroofficial — APPROVED** [review]
This look good.

File: src/test/components/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import HomePage from '../../components/HomePage';
+
+configure({ adapter: new Adapter() });
+
+describe('<Home>', () => {
+  it('should have a p tag', () => {
+    // eslint-disable-next-line react/react-in-jsx-scope
+    // eslint-disable-next-line react/jsx-filename-extension
+    const wrapper = shallow(<HomePage />);
+    console.log(wrapper.debug());
```
Comment (fejiroofficial) [inline src/test/components/Home.test.js:14]: #### Nitpick
Why are you logging wrapper.debug()

File: src/test/components/Home.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { configure, shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import HomePage from '../../components/HomePage';
+
+configure({ adapter: new Adapter() });
```
Comment (aanchirinah) [inline src/test/components/Home.test.js:7]: #### Nitpick 
- Isn't there a better way to do this?

**Review by aanchirinah — APPROVED** [review]
LGTM

**Pull request merged**